doc/source/security/data-masking.rst

```diff
@@ -0,0 +1,307 @@
+            | mask_outer("123456789", 2, 2).     |
+            +------------------------------------+
+            | XX34567XX                          |
+            +------------------------------------+
+
+.. rubric:: Special Purpose
+
+The special purpose data masking functions are as follows:
+
+.. tabularcolumns:: |p{0.25\linewidth}|p{0.25\linewidth}|p{0.50\linewidth}|
+
+.. list-table::
+    :widths: 20 30 60
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+      - Sample
+
+    * - mask_pan(string)
+      - Masks the Primary Account Number (PAN) by replacing the
+        string with an "X" except for the last four characters.
+
+        .. note::
+
+            The PAN string must be 15 characters or 16 characters in length.
+
+      - .. code-block:: MySQL
+
+            mysql> SELECT mask_pan ('123456789012345');
+
+            +------------------------------------+
+            | mask_pan(gen_rnd_pan()).           |
+            +------------------------------------+
+            | XXXXXXXXXXX2345                    |
+            +------------------------------------+
+
+    * - mask_pan_relaxed(string)
+      - Returns the first six numbers and the last four    numbers. The rest of
+        the string is replaced by "X".
+
+      - .. code-block:: MySQL
+
+            mysql> SELECT mask_pan_relaxed(gen_rnd_pan());
+
+            +------------------------------------------+
+            | mask_pan_relaxed(gen_rnd_pan())          |
+            +------------------------------------------+
+            | 520754XXXXXX4848                         |
+            +------------------------------------------+
+    * - mask_ssn(string)
+      - Returns a  string with only the last four numbers visible. The rest
+        of the string is replaced by "X".
+
+      - .. code-block:: MySQL
+
+            mysql> SELECT mask_ssn('555-55-5555');
+
+            +-------------------------+
+            | mask_ssn('555-55-5555') |
+            +-------------------------+
+            | XXX-XX-5555             |
+            +-------------------------+
+
+.. rubric:: Generating Random Data for Specific Requirements
+
+The following functions generate random values for specific requirements:
+
+.. tabularcolumns:: |p{0.25\linewidth}|p{0.25\linewidth}|p{0.50\linewidth}|
+
+.. list-table::
+    :widths: 20 30 60
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+      - Sample
+    * - gen_range(lower, upper)
+      - Generates a random number based on a selected range and supports 
+        negative numbers.
+
+      - .. code-block:: MySQL
+
+              mysql> SELECT gen_range(10, 100);
+
+              +--------------------------------------+
+              | gen_range(10,100)                    |
+              +--------------------------------------+
+              | 56                                   |
+              +--------------------------------------+
+
+              mysql> SELECT gen_range(-100,-80);
```

Comment:
Are you sure this worked for negative numbers?

Reply:
PS 7309 says that it does.

Reply:
@patrickbirch Right, but that was fixed only in 8.0.
IMO, we should push this PR only to 8.0.

5.7, we should backport the bugfixes and then backport this documentation ticket. What do u say?